packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -335,7 +335,8 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
           url = resolved.id.slice(root.length)
         } else if (
           depsOptimizer?.isOptimizedDepFile(resolved.id) ||
-          fs.existsSync(cleanUrl(resolved.id))
+          (path.isAbsolute(cleanUrl(resolved.id)) &&
+            fs.existsSync(cleanUrl(resolved.id)))
         ) {
```

Comment:
`resolved.id` can be `pkg/foo.js` and in that case we should not append `/@fs/` as it's not an absolute path.